### PR TITLE
add documents about multi match and match phrase

### LINKED ISSUE #3728: Add documentation for the ES multimatch query and the phrase match query.
Documentation page: [es_compatible_api.md](https://github.com/quickwit-oss/quickwit/blob/main/docs/reference/es_compatible_api.md)

Multi match query: [Elastic ref](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-multi-match-query.html)

Match phrase query: [Elastic ref](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-match-query-phrase.html)

TODO: 
- [ ] Add multi-match query docs in `es_compatible_api.md`
- [ ] Add match phrase query docs in `es_compatible_api.md`
- [ ] Check endpoints on a small dataset

### PR DESCRIPTION
**Pull request by JerryKwan**
add documents about multi match query and match phrase query

### Description

add documents about multi match query and match phrase query
fix link error in match phrase query

### How was this PR tested?

Describe how you tested this PR.

Closes #3728

### CODE REVIEW

File: docs/reference/es_compatible_api.md [hunk 1]
```diff
@@ -342,6 +342,23 @@ The following query types are supported.
...
+#### Example
+
+```json
+{
+  "query": {
+    "match_phrase": {
+      "title": "search keywords"
+    }
+  }
+}
```
Comment (fmassot) [inline docs/reference/es_compatible_api.md:358]: Just to show we support the analyzer field too:

```suggestion
{
  "query": {
    "match_phrase": {
      "title": "search keywords",
      "analyzer": "default"
    }
  }
}
```

File: docs/reference/es_compatible_api.md [hunk 2]
```diff
@@ -398,6 +415,70 @@ Contrary to ES/Opensearch, in Quickwit, at most 50 terms will be considered when
...
+    "multi_match" : {
+      "query":      "search key",
+      "type":       "phrase_prefix",
+      "fields":     [ "title", "body" ]
+    }
+  }
+}
+```
+
+#### Supported Multi_match Queries
```
Comment (fmassot) [inline docs/reference/es_compatible_api.md:472]:
```suggestion
#### Supported Multi-match Queries
```

File: docs/reference/es_compatible_api.md [hunk 3]
```diff
@@ -398,6 +415,70 @@ Contrary to ES/Opensearch, in Quickwit, at most 50 terms will be considered when
...
+| Type    | Description                                                                              |
+|---------|------------------------------------------------------------------------------------------|
+| `most_fields` | (default) Finds documents which match any field and combines the _score from each field. |
+| `phrase` | Runs a match_phrase query on each field and uses the _score from the best field          |
```
Comment (fmassot) [inline docs/reference/es_compatible_api.md:476]:
```suggestion
| `phrase` | Runs a `match_phrase` query on each field and uses the `_score` from the best field          |
```

File: docs/reference/es_compatible_api.md [hunk 4]
```diff
@@ -398,6 +415,70 @@ Contrary to ES/Opensearch, in Quickwit, at most 50 terms will be considered when
...
+| `phrase_prefix` | Runs a match_phrase_prefix query on each field and uses the _score from the best field   |
```
Comment (fmassot) [inline docs/reference/es_compatible_api.md:477]:
```suggestion
| `phrase_prefix` | Runs a `match_phrase_prefix` query on each field and uses the `_score` from the best field   |
```

**Review by fmassot — APPROVED** [review]
LGTM, can you check the comments? (it's only very minor things)

**Comment by JerryKwan** [comment]
@fmassot Updated
@fmassot @fulmicoton Thanks for reviewing

**Pull request merged**